lib/engine/game/g_rolling_stock/game.rb

```diff
@@ -0,0 +1,194 @@
+             9
+             10
+             11
+             12
+             13
+             14
+             16
+             18
+             20
+             22
+             24
+             27
+             30
+             33
+             37
+             41
+             45
+             50
+             55
+             61
+             68
+             75e],
+        ].freeze
+
+        MARKET_TEXT = {
+          par: 'Par value',
+          no_cert_limit: 'Corporation shares do not count towards cert limit',
+          unlimited: 'Corporation shares can be held above 60%',
+          multiple_buy: 'Can buy more than one share in the corporation per turn',
+          close: 'Corporation closes',
+          endgame: 'End game trigger',
+          liquidation: 'Liquidation',
+          repar: 'Minor company value',
+          ignore_one_sale: 'Ignore first share sold when moving price',
+        }.freeze
+
+        TILES = [].freeze
+        CERT_LIMIT = 99
+
+        SELL_MOVEMENT = :left_share_pres
+        SELL_BUY_ORDER = :sell_buy
+        GAME_END_CHECK = { stock_market: :current_or, bank: :full_or }.freeze
+        SOLD_OUT_INCREASE = false
+        EBUY_OTHER_VALUE = false
+
+        PHASES = [
+          { name: 'red', train_limit: 1, tiles: [:yellow], operating_rounds: 1 },
+          { name: 'orange', train_limit: 1, tiles: [:yellow], operating_rounds: 1 },
+          { name: 'yellow', train_limit: 1, tiles: [:yellow], operating_rounds: 1 },
+          { name: 'green', train_limit: 1, tiles: [:yellow], operating_rounds: 1 },
+          { name: 'blue', train_limit: 1, tiles: [:yellow], operating_rounds: 1 },
+        ].freeze
+
+        def num_to_draw(players, stars)
+          if players.size == 6
+            8
+          elsif stars != 2 || players.size < 4
+            players.size
+          elsif players.size == 4
+            5
+          else
+            7
+          end
+        end
+
+        def setup
+          @company_stars = self.class::COMPANIES.to_h { |c| [c[:sym], c[:stars]] }
+          @company_synergies = self.class::COMPANIES.to_h { |c| [c[:sym], c[:synergies]] }
+
+          @company_deck = []
+          5.times do |stars|
+            matching = @companies.select { |c| @company_stars[c.sym] == (stars + 1) }
+            highest = matching.max_by(&:value)
+            drawn = matching.reject { |c| c == highest }.sort_by { rand }.take(num_to_draw(players, stars + 1))
+            drawn << highest
+            @company_deck.concat(drawn.sort_by { rand })
+          end
+          @offering = @company_deck.shift(@players.size)
+          @on_deck = []
+        end
+
+        def init_round
+          new_investment_round
+        end
+
+        def new_investment_round
+          @log << "-- Turn #{@turn}, Phase 1 - Investment --"
+          @round_counter += 1
+          investment_round
+        end
+
+        def investment_round
+          GRollingStock::Round::Investment.new(self, [
```

Comment:
do you have to prefix with GRollingStock, isn't that implicit?

Reply:
I thought I needed to, but apparently not.